utils/src/userInput/CopilotUserInput.ts

```diff
@@ -0,0 +1,118 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+import type * as vscodeTypes from 'vscode';
+import { workspace } from 'vscode';
+import * as types from '../../index';
+import { createPrimaryPromptForInputBox, createPrimaryPromptForWarningMessage, createPrimaryPromptForWorkspaceFolderPick, createPrimaryPromptToGetPickManyQuickPickInput, createPrimaryPromptToGetSingleQuickPickInput, doCopilotInteraction } from '../copilot/copilot';
+import { InvalidInputError } from '../errors';
+
+export class CopilotUserInput implements types.IAzureUserInput {
+    private readonly _vscode: typeof vscodeTypes;
+    private readonly _onDidFinishPromptEmitter: vscodeTypes.EventEmitter<types.PromptResult>;
+    private readonly _relevantContext: string | undefined;
+
+    constructor(vscode: typeof vscodeTypes, relevantContext?: string) {
+        this._vscode = vscode;
+        this._onDidFinishPromptEmitter = new this._vscode.EventEmitter<types.PromptResult>();
+        this._relevantContext = relevantContext;
+    }
+
+    public async showWarningMessage<T extends vscodeTypes.MessageItem>(message: string, ...items: T[]): Promise<T> {
+        const primaryPrompt: string = createPrimaryPromptForWarningMessage(message, items);
+        const response = await doCopilotInteraction(primaryPrompt)
+
+        const pick = items.find(
+            item => {
+                return item.title === response;
+            }
+        );
+
+        if (!pick) {
+            throw new InvalidInputError();
+        }
+
+        this._onDidFinishPromptEmitter.fire({ value: pick });
+        return pick;
+    }
+
+    public showOpenDialog(): Promise<vscodeTypes.Uri[]> {
+        // Throw this back to the user
+        throw new InvalidInputError();
```

Comment:
How jarring is it when the error gets thrown back and the user has to take control? I'm not sure if there's a great way to do it, but just curious what it currently looks like.

Reply:
Does the webview that says Copiloting is generating the picks close?

Reply:
I don't think so at the moment. Since the webview lives only in the aca extension there isn't that good of a way to dispose of it here. I may can add a listener on the ACA side to see if I can dispose of it anytime that error is thrown. The view does get disgarded once the picks are all finished but yeah I totally forgot about that.